satpy/composites/__init__.py

```diff
@@ -1908,6 +1908,9 @@ def __call__(self, projectables, *args, **kwargs):
         data_in = projectables[0]
         mask_in = projectables[1]
 
+        mask_in = mask_in.squeeze(drop=True)
```

Comment:
I think this solution is fine for now, but there may be a longer term solution of checking for a "bands" dimension of size 1 and selecting that single band. By doing squeeze I'm worried that this breaks conditions where input arrays are more than 2 dimensions but don't only include the "bands" dimension. This is unlikely in current Satpy but still.